eis_toolkit/vector_processing/simple_idw.py

```diff
@@ -0,0 +1,113 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+from beartype import beartype
+from beartype.typing import Optional, Tuple, Union
+
+from eis_toolkit.exceptions import EmptyDataFrameException, InvalidParameterValueException
+
+
+@beartype
+def _simple_idw(
+    geodataframe: gpd.GeoDataFrame,
+    target_column: str,
+    resolution: Tuple[Number, Number],
+    power: Number,
+    extent: Union[Tuple[Number, Number, Number, Number], None],
```

Comment:
Although this is equivalent with `Optional[Tuple[Number, Number, Number, Number]]` in simple_idw, I'd declare the parameter type the same way in both functions for consistency.
```suggestion
    extent: Optional[Tuple[Number, Number, Number, Number]]
```